### ISSUE TITLE
Other Estimators and trainers

### ISSUE THREAD

**Issue by nfnpmc** [dotnet/docs#13869]
Any thing other than mlContext.Regression.Trainers?
Give us a clue.

[AB#1579020](https://dev.azure.com/mseng/677da0fb-b067-4f77-b89b-f32c12bb8617/_workitems/edit/1579020)

---
#### Document details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 784daf15-838b-a41d-98f3-779ac307745f
* Version Independent ID: ddc8c98a-f8f8-f35b-0677-4c8f77c54236
* Content: [Save and load trained models - ML.NET](https://docs.microsoft.com/en-gb/dotnet/machine-learning/how-to-guides/save-load-machine-learning-models-ml-net)
* Content Source: [docs/machine-learning/how-to-guides/save-load-machine-learning-models-ml-net.md](https://github.com/dotnet/docs/blob/master/docs/machine-learning/how-to-guides/save-load-machine-learning-models-ml-net.md)
* Product: **dotnet-ml**
* GitHub Login: @luisquintanilla
* Microsoft Alias: **luquinta**

**Comment by luisquintanilla** [comment]
Hi @nfnpmc ,

Thank you for your question. Regression is just one of the many model types supported by ML.NET. This sample is illustrating how to load / save models which is supported by the other model types. You may find the following resources helpful:

https://docs.microsoft.com/en-us/dotnet/machine-learning/how-to-choose-an-ml-net-algorithm

https://docs.microsoft.com/en-us/dotnet/api/microsoft.ml.trainers?view=ml-dotnet

https://docs.microsoft.com/en-us/dotnet/machine-learning/tutorials/

**Comment by nfnpmc** [comment]
Hi Luis,Thanks for your prompt response.I guess I wasn't very specific. I was looking at 
Working with separate data preparation and model pipelines
My estimator chain (called pipline) is for a Tensorflow model and if I doTransformerChain<Microsoft.ML.Transforms.KeyToValueMappingTransformer> transformer = pipeline.Fit(data);I think I end up with a trained model (transformer )?
If not and I wanted to do 
Microsoft.ML.Trainers.SdcaRegressionTrainer sdcaEstimator = context.Regression.Trainers.Sdca();
            RegressionPredictionTransformer<LinearRegressionModelParameters> trainedModel = sdcaEstimator.Fit(transformedData);
What Trainer should I use?Aha! I have just spotted "LbfgsMaximumEntropy"
.Append(context.Transforms.ExtractPixels(outputColumnName: "input",
                                                         interleavePixelColors: ImageNetSettings.channelsLast,
                                                         offsetImage: ImageNetSettings.mean))
.Append(tfModel.ScoreTensorFlowModel(outputColumnNames: new[] { "softmax2_pre_activation" },
                                                     inputColumnNames: new[] { "input" },
                                                     addBatchDimensionInput: true))
.Append(context.MulticlassClassification.Trainers.LbfgsMaximumEntropy(labelColumnName: "Label",
                                                                                      featureColumnName: "softmax2_pre_activation"))
.Append(context.Transforms.Conversion.MapKeyToValue("Prediction", "PredictedLabel"));

so, Microsoft.ML.Trainers.LbfgsMaximumEntropyMulticlassTrainer Estimator = context.MulticlassClassification.Trainers.LbfgsMaximumEntropy();
MulticlassPredictionTransformer<Microsoft.ML.Trainers.MaximumEntropyModelParameters> trainedModel = Estimator.Fit(transformedData);
Do I have it? 
So now to what I was trying to achieve:Save a system and reload it so I can train it again on different data.And again ...

When I load them I will end up with 2 ITransformers.
// Load data preparation pipeline and trained model
ITransformer dataPrepPipeline = mlContext.Model.Load("data_preparation_pipeline.zip",out dataPrepPipelineSchema);
ITransformer trainedModel = mlContext.Model.Load("model.zip", out modelSchema);I will try it out, thank you.

Regards, Peter 

    On Friday, 16 August 2019, 15:54:43 BST, Luis Quintanilla <notifications@github.com> wrote:  
 
 
Hi @nfnpmc ,

Thank you for your question. Regression is just one of the many model types supported by ML.NET. This sample is illustrating how to load / save models which is supported by the other model types. You may find the following resources helpful:

https://docs.microsoft.com/en-us/dotnet/machine-learning/how-to-choose-an-ml-net-algorithm

https://docs.microsoft.com/en-us/dotnet/api/microsoft.ml.trainers?view=ml-dotnet

https://docs.microsoft.com/en-us/dotnet/machine-learning/tutorials/
